packages/strapi-plugin-upload/admin/src/components/InputMedia/index.js

```diff
@@ -14,12 +16,21 @@ import Name from './Name';
 import Wrapper from './Wrapper';
 
 const InputMedia = ({ label, onChange, name, attribute, value, type }) => {
-  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modal, setModal] = useState({
+    isOpen: false,
+    step: null,
+  });
   const [fileToDisplay, setFileToDisplay] = useState(0);
+  const hasNoValue = value !== null && Array.isArray(value) && value.length === 0;
```

Comment:
`const hasNoValue = !!value && Array.isArray(value) && value.length === 0;`  so you can handle the `undefined` case.